File: lib/models/appointment.dart
class Appointment {
  final String id;            // unique id
  final String doctorName;
  final String location;
  final DateTime date;
  final String slot;
  final String type;          // 'In-Person' | 'Virtual'
  final String notes;
  final String status;        // 'Scheduled' | 'Cancelled'

  Appointment({
    required this.id,
    required this.doctorName,
    required this.location,
    required this.date,
    required this.slot,
    required this.type,
    required this.notes,
    this.status = 'Scheduled',
  });

  Appointment copyWith({
    String? id,
    String? doctorName,
    String? location,
    DateTime? date,
    String? slot,
    String? type,
    String? notes,
    String? status,
  }) {
    return Appointment(
      id: id ?? this.id,
      doctorName: doctorName ?? this.doctorName,
      location: location ?? this.location,
      date: date ?? this.date,
      slot: slot ?? this.slot,
      type: type ?? this.type,
      notes: notes ?? this.notes,
      status: status ?? this.status,
    );
  }
}

File: backend/src/models/appointment.js
const mongoose = require('mongoose');

const appointmentSchema = new mongoose.Schema({
  patient: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Patient",
    required: true,
  },
  doctor: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Doctor",
    required: true,
  },
  appointmentDate: {
    type: Date,
    required: true,
  },
  symptoms: {
    type: String,
    default: "",
  },
  status: { //NEED TO IMPLEMENT
    type: String,
    // enum: ["pending", "scheduled", "reScheduled", "cancelled"],
    default: "pending",
  },
  appointmentType: {
    type: String,
    enum: ["in-person", "virtual", "telehealth"],
    default: "in-person",
  },
  notes: {
    type: String,
  },
  selectedTimeSlot: {
    from: { type: String, required: true }, // e.g. "09:00"
    to: { type: String, required: true },   // e.g. "10:00"
    // period: { type: String, enum: ["AM", "PM"], required: true }, // "AM" or "PM"
    availabilityStatus: { type: Boolean, required: true }
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

const Appointment = mongoose.model("Appointment", appointmentSchema);
module.exports = Appointment;
